resources/migrations/0074_alter_resource_fields.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.18 on 2019-02-11 11:27
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('resources', '0073_add_resource_generate_access_codes_field'),
+    ]
+
+    operations = [
```

Comment:
Have you tested what happens if the database contains negative numbers in these fields when executing the migration?

To be safe you could add an operation here which is executed first and sets all negative values to 0 in these fields. Then the alter columns will surely succeed.